pkg/providers/aws/provider_postgres.go

```diff
@@ -0,0 +1,487 @@
+package aws
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/google/uuid"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/rds/rdsiface"
+	"github.com/integr8ly/cloud-resource-operator/pkg/resources"
+
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/integr8ly/cloud-resource-operator/pkg/apis/integreatly/v1alpha1"
+	"github.com/integr8ly/cloud-resource-operator/pkg/providers"
+	"github.com/sirupsen/logrus"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	errorUtil "github.com/pkg/errors"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	postgresProviderName       = "aws-rds"
+	defaultAwsIdentifierLength = 40
+	// default create options
+	defaultAwsPostgresDeletionProtection = true
+	defaultAwsPostgresPort               = 5432
+	defaultAwsPostgresUser               = "postgres"
+	defaultAwsAllocatedStorage           = 20
```

Comment:
According to the [docs](https://docs.aws.amazon.com/sdk-for-go/api/service/rds/#CreateDBInstanceInput) it will not scale without the max allocated storage set. 
Aurora has the feature for auto scaling without setting max allocated storage.

Could this be a candidate to bring out of the strat and move to the spec?